packages/validator/src/services/chainHeaderTracker.ts

```diff
@@ -31,8 +31,10 @@ export class ChainHeaderTracker {
   ) {}
 
   start(signal: AbortSignal): void {
-    void this.api.events.eventstream([EventType.head], signal, this.onHeadUpdate);
-    this.logger.verbose("Subscribed to head event");
+    this.logger.verbose("Subscribing to head event");
+    this.api.events
+      .eventstream([EventType.head], signal, this.onHeadUpdate)
+      .catch((e) => this.logger.error("Failed to subscribe to head event", {}, e));
```

Comment:
Unrelated change but noticed that currently errors result in an uncaught exception

Reply:
Caller won't receive the exception anymore, so any code after the call to `start` will be executed now.
I understand from your comment that it's fine?

Reply:
The caller never received the error as the promise was not handled (see `void` in previous implementation).

The `eventstream` method does not resolve until connection is aborted which effectively happens if you shut down the validator client
https://github.com/ChainSafe/lodestar/blob/959a8af1b757c46a0e57450788fb72773a372608/packages/api/src/beacon/client/events.ts#L48-L49

But there are cases where it rejects if server responds with 400 or 500 error
https://github.com/ChainSafe/lodestar/blob/959a8af1b757c46a0e57450788fb72773a372608/packages/api/src/beacon/client/events.ts#L36

Previously due to `void`, these errors would cause an uncaught exception while now the erros are caught and properly logged